cobra/io/sbml3.py

```diff
@@ -548,7 +548,13 @@ def read_sbml_model(filename, number=float, **kwargs):
                 xmlfile.write(outfile, encoding="UTF-8")
             filename = outfile.name
         return read_sbml2(filename, **kwargs)
-    return parse_xml_into_model(xml, number=number, **kwargs)
+    try:
+        return parse_xml_into_model(xml, number=number, **kwargs)
+    except Exception:
```

Comment:
I can add a test for the new try/except and maybe add an id checker to `parse_xml_model` until tomorrow if that's fine.